workers/common/src/main/kotlin/common/env/ConfigFileBuilder.kt

```diff
@@ -77,6 +82,42 @@ class ConfigFileBuilder(val context: WorkerContext) {
             }
         }
 
+        /**
+         * Execute the given [block] to generate a proxy configuration if necessary.
+         * This variant is based on system properties that define https(s) proxy settings rather than
+         * environment variables. It is typically used by tools that have a strong connection to the Java ecosystem.
+         * If one of httpProxyHost or httpsProxyHost is defined, the [block] is invoked with a corresponding
+         * [ProxiesFromSystemProperties].
+         *
+         * @param block A lambda function that is an extension function on PrintWriter and takes a
+         * [ProxiesFromSystemProperties] object as a parameter.
+         */
+        fun PrintWriter.printProxySettingsFromSystemProperties(
+            block: PrintWriter.(ProxiesFromSystemProperties) -> Unit
+        ) {
+            val httpProxy = HTTP_PROXY_HOST.getValue()?.let { host ->
```

Comment:
Since there are now multiple enumerations involved, I would find it more readable if the constants were qualified, i.e. `ProxySystemProperties.HTTP_PROXY_HOST` etc.

Reply:
Yes, added ProxyStemProperties as proposed.